### DISCUSSION TITLE
Generate a special class on image links?

### DISCUSSION THREAD

**Discussion by berezovskyi** [getpelican/pelican#3171] [Q&A]
I am trying to style images a bit unconventionally, which conflicts with link styles when an image is wrapped with a link (`a > img`).

I tried to override some of the link styles if a link contains an img using the `a:has(> img)` selector, but I discovered that Firefox does not support it.

What I would like is to have the generator produce the generate the following HTML from text links and image links:

```html
<a href="xyz">abc</a>
<a href="xyz" class="imagelink"><img src="xyz2"></a>
```

Is it possible to do with Pelikan? Here are my dependency versions:

```
pelican==4.8.0
Jinja2==3.1.2
Markdown==3.4.1
smartypants==1.8.6
typogrify==2.0.7
pelican-tag-cloud==1.0.1
```

Thank you in advance!

**Comment by justinmayer** [comment]
Pelican hands your source content to the respective library for that source content. You didn't specify what format your source content is, but for the moment I will assume it is in Markdown format. In that case, Pelican hands your source content to [Python-Markdown](https://github.com/Python-Markdown/markdown), and thus Pelican itself has no control over how that library converts Markdown to HTML.

It is possible that there may be a [Python-Markdown](https://github.com/Python-Markdown/markdown) extension that can help you accomplish your objective: https://github.com/Python-Markdown/markdown/wiki/Third-Party-Extensions

Alternatively, there are Pelican plugins that relate to your target objective, although I suspect neither will achieve precisely what you want in their current form:

- https://github.com/pelican-plugins/injector
- https://github.com/pelican-plugins/linkclass

One approach could be to review the source code for the Injector plugin, modify it appropriately to support a _**generalized**_ use case that also addresses your specific need, and then submit a pull request to that repository for inclusion in that plugin.

**Reply by berezovskyi** [reply]
Thank you! I will look into the `injector` plugin and see if I can extend it with some generic class injection to an element or class injection to the parent.

For now, I [enabled `markdown.extensions.extra`](https://jackdewinter.github.io/2019/10/16/fine-tuning-pelican-markdown-configuration/#markdown-configuration) and changed my markdown links to `[![]({filename}/xyz)](abc){: .imagelink}` (so, not generated, unfortunately). I have RST sources too but seems like I never wrapped figures with links there.
